packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -329,6 +335,16 @@ class PigeonOptions {
   PigeonOptions merge(PigeonOptions options) {
     return PigeonOptions.fromMap(mergeMaps(toMap(), options.toMap()));
   }
+
+  /// Returns provided or deduced package name, throws `Exception` if none found.
+  String getPackageNameOrThrow() {
```

Comment:
Generally Dart exceptions aren't encoded into the name; this can just be `getPackageName`.